### PR TITLE
comment replies

### LINKED ISSUE #2443: Reply to comments
Reply to comments [link](https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2213%3A140576)

- [ ] Thread
- [ ] All replies are always sorted by their publication date, starting with the newest ones.

Out of scope:
- [ ] ~Subthread~

covered in https://github.com/Joystream/atlas/issues/2522

### PR DESCRIPTION
**Pull request by rafalpawlow**
Closes #2443 

**Design**
https://www.figma.com/file/sMydmZJlzhK91FTIPOd7D2/Video-page?node-id=2302%3A285300

### CODE REVIEW

**Comment by eldiegod** [comment]
needs rebase again, i know the pain 🥲

**Comment by rafalpawlow** [comment]
@eldiegod  Argh, I forgot. On my way!

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: Maybe add a comment to what this is, also is not clear what initialFocus is

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (eldiegod) [inline packages/atlas/src/views/viewer/VideoView/CommentThread.tsx]: dont think this if statement is necessary

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (eldiegod) [inline packages/atlas/src/views/viewer/VideoView/CommentThread.tsx:82]: wrapping this in useCallback is unnecessary because where it's used the handler is created every render anyways

File: packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx
Comment (eldiegod) [inline packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx]: the main comment input in the video page doesn't use `onCancel` or has a cancel button on design

**Review by eldiegod — CHANGES_REQUESTED** [review]
Mostly good stuff with some details

- Replies don't seem to be sorted by the newest first

**Review by drillprop — CHANGES_REQUESTED** [review]
I've noticed some issues:
- You cannot click the reply button on the comment which is a reply. Nothing happens, but the button is still visible. 
- If you click reply on one comment and you write something in comment input(but don't send a new comment), you need to click two times on the other comment's reply button to open new comment input. I hope it makes sense :P
- When the input is empty and you click cancel, the "discard changes" dialog is showing.  I don't think this should work that way. 
- I think when the input is open and the user clicks the reply button again(to the same comment) we should focus on that input.

**Comment by rafalpawlow** [comment]
@drillprop Reply to reply is out of scope https://github.com/Joystream/atlas/issues/2522

**Comment by drillprop** [comment]
> @drillprop Reply to reply is out of scope #2522

Do we have an issue created for a reply to reply? Do we know how it will work in the future?

**Comment by rafalpawlow** [comment]
@drillprop Yes it's linked in my comment 😛

**Comment by rafalpawlow** [comment]
@eldiegod sorting is correct, there is a mistake in the user story and issue description. I confirmed it with @toiletgranny, he will correct descriptions.

EDIT: https://github.com/Joystream/atlas/issues/2443#issuecomment-1129659033

**Comment by drillprop** [comment]
> @drillprop Yes it's linked in my comment 😛

oh god 🤦🏻‍♂️

**Review by drillprop — CHANGES_REQUESTED** [review]
One more thing, I'm not sure if I missed that before or if you just introduced this. 
I don't see my avatar in comment input 
image

**Review by drillprop — APPROVED** [review]
As for me, you got my approval, works perfectly now! 
I only reviewed and tested functionality though, didn't look at the code at all. Let's wait for final approval from Diego

File: packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx
Comment (eldiegod) [inline packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx]: we also needed to remove preventDefault but I do it in my PR as well so no worries

File: packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx
Comment (eldiegod) [inline packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx]: This fn doesn't need to be in this component and be passed as a prop; we can just move it down to `CommentThread`

**Review by eldiegod — APPROVED** [review]
Working good

**Review by kdembler — CHANGES_REQUESTED** [review]
There is a reply button for replies but when I click it, nothing happens

**Review by drillprop — CHANGES_REQUESTED** [review]
Found one more thing. Reactions not working correctly under the reply: 
- The onboarding popover is popping up randomly(?)
- Sometimes you don't see a processing reaction once you click reaction on the reaction popover. 

Edit: 
Actually only ReactionPopover doesn't work. When you click on reaction in ReactionPopover onboarding popover is always showing up. And when you click reaction, the reaction is not changing state to `processing`. 

Everything is working when you click on the reaction chip(existing reaction)

**Review by drillprop — APPROVED** [review]

**Comment by rafalpawlow** [comment]
@eldiegod @drillprop I need to ask you to review it one more time because of new tweaks and fixes

File: packages/atlas/src/components/_comments/Comment/Comment.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/Comment/Comment.tsx]: This will not work if two users have the same avatar. 
Also, I think we could do this a little better. There is a nice trick that will remove duplicates. You could do this at line 98
```js
  const avatarsWithoutDuplicates = [...new Map(replyAvatars?.map((item) => [item.handle, item])).values()]
```

File: packages/atlas/src/components/_comments/Comment/Comment.tsx
Comment (drillprop) [inline packages/atlas/src/components/_comments/Comment/Comment.tsx]:
```  
const avatarsWithoutDuplicates = [...new Map(replyAvatars?.map((item) => [item.handle, item])).values()]
```

**Review by drillprop — CHANGES_REQUESTED** [review]

**Comment by eldiegod** [comment]
will need a rebase

**Review by kdembler — CHANGES_REQUESTED** [review]
1. I cannot post a reply, the `Comment` button seems to be disabled at all times?<br>Screenshot 2022-05-23 at 09 41 00
2. For comment replies, all the reactions are in "pressed" state, like I reacted to those myself, when that's not the case. This also happens even when I'm not logged in

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (kdembler) [inline packages/atlas/src/views/viewer/VideoView/CommentThread.tsx:68]: `useMemberAvatar` should be used for this, we need to get rid of this pattern

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (kdembler) [inline packages/atlas/src/views/viewer/VideoView/CommentThread.tsx]: Wrong prop name

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (kdembler) [inline packages/atlas/src/views/viewer/VideoView/CommentThread.tsx]: `useMemberAvatar` should be used for this, we need to get rid of this pattern

**Review by kdembler — CHANGES_REQUESTED** [review]

File: packages/atlas/src/api/hooks/comments.ts
Comment (drillprop) [inline packages/atlas/src/api/hooks/comments.ts]: Can we also apply [this function](https://github.com/Joystream/atlas/pull/2721/files#diff-7bfe1bb30c6e04890de539e112ce9ae220fd47a53fe944866684d64c3b511bbbR28) here?

File: packages/atlas/src/components/_comments/Comment/Comment.styles.ts
Comment (drillprop) [inline packages/atlas/src/components/_comments/Comment/Comment.styles.ts]: Hmm. I'm curious. Why do we need to additional 1px top? I also noticed that you also applied `align-items: center` on the wrapper, but it looks like it does nothing, can you double-check why this is not working? 

Just to clarify - If there is no easy solution to get rid of this I'm fine with leaving it as it is now.

**Review by drillprop — CHANGES_REQUESTED** [review]

File: packages/atlas/src/views/viewer/VideoView/CommentThread.tsx
Comment (rafalpawlow) [reply packages/atlas/src/views/viewer/VideoView/CommentThread.tsx:68]: This will be resolved here https://github.com/Joystream/atlas/issues/2735

**Review by kdembler — APPROVED** [review]
Will need some cleanup as part of @eldiegod PR but I think it looks good and works well 👍

**Review by drillprop — APPROVED** [review]
LGTM as well!

**Pull request merged**
